test/unit/org/apache/cassandra/db/commitlog/CommitLogBytemanTest.java

```diff
@@ -0,0 +1,173 @@
+        KeyspaceParams.DEFAULT_LOCAL_DURABLE_WRITES = false;
+
+        SchemaLoader.prepareServer();
+        StorageService.instance.getTokenMetadata().updateHostId(UUID.randomUUID(), FBUtilities.getBroadcastAddressAndPort());
+
+        MemtableParams skipListMemtable = MemtableParams.fromMap(ImmutableMap.of("class", "SkipListMemtable"));
+
+        TableMetadata.Builder custom =
+                TableMetadata.builder(KEYSPACE, CUSTOM)
+                        .addPartitionKeyColumn("k", IntegerType.instance)
+                        .addClusteringColumn("c1", MapType.getInstance(UTF8Type.instance, UTF8Type.instance, false))
+                        .addClusteringColumn("c2", SetType.getInstance(UTF8Type.instance, false))
+                        .addStaticColumn("s", IntegerType.instance)
+                        .memtable(skipListMemtable);
+
+        SchemaLoader.createKeyspace(KEYSPACE,
+                KeyspaceParams.simple(1),
+                SchemaLoader.standardCFMD(KEYSPACE, STANDARD, 0, AsciiType.instance, BytesType.instance).memtable(skipListMemtable),
+                custom);
+        CompactionManager.instance.disableAutoCompaction();
+
+        testKiller = new KillerForTests();
+
+        // While we don't want the JVM to be nuked from under us on a test failure, we DO want some indication of
+        // an error. If we hit a "Kill the JVM" condition while working with the CL when we don't expect it, an aggressive
+        // KillerForTests will assertion out on us.
+        oldKiller = JVMStabilityInspector.replaceKiller(testKiller);
+    }
+
+    @AfterClass
+    public static void afterClass()
+    {
+        JVMStabilityInspector.replaceKiller(oldKiller);
+    }
+
+    @Before
+    public void beforeTest() throws IOException
+    {
+        CommitLog.instance.resetUnsafe(true);
+    }
+
+    @After
+    public void afterTest()
+    {
+        testKiller.reset();
+    }
+
+    @Test
+    @BMRules(rules = { @BMRule(name = "Fail sync in CommitLog",
+            targetClass = "CommitLog",
+            targetMethod = "sync",
+            condition = "org.apache.cassandra.db.commitlog.CommitLogBytemanTest.failSync",
+            action = "throw new java.lang.RuntimeException(\"Fail CommitLog.sync to test fail_writes policy\");") } )
+    public void testFailWritesPolicies() throws InterruptedException
+    {
+        ColumnFamilyStore cfs = Keyspace.open(KEYSPACE).getColumnFamilyStore(STANDARD);
+        Config.CommitFailurePolicy oldPolicy = DatabaseDescriptor.getCommitFailurePolicy();
+        DatabaseDescriptor.setCommitFailurePolicy(Config.CommitFailurePolicy.fail_writes);
+
+        Mutation m = new RowUpdateBuilder(cfs.metadata.get(), 0, "key")
+                .clustering("bytes")
+                .add("val", ByteBuffer.allocate(10 * 1024))
+                .build();
+        CommitLog.instance.add(m);
+
+        Thread awaitForSync = new Thread(CommitLogBytemanTest.class.getSimpleName() + " commit log waiting thread")
```

Comment:
probably add a comment that `        CommitLog.instance.add(m);` waits for sync because in tests we're using `BatchCommitLogService`?

even better - add an assertion that we do, and treat it is a precondition for this test (with a nice message describing why)

Reply:
Finally I got to work the group CL service, so it is not blocking and no additional thread is needed.

Reply:
So the test looks now as you proposed.